robotbase/src/main/java/frc/robot/subsystems/Shooter.java

```diff
@@ -103,14 +114,11 @@ public void periodic() {
     }
   }
 
-  public void startVisionShooting() {
-    m_isClosedLoopEnabled = true;
-    Robot.shooterLimelight.setPipeline(Constants.SHOOTER_LIMELIGHT.SPEAKER_PIPELINE_INDEX);
-  }
-
-  public void endVisionShooting() {
-    m_isClosedLoopEnabled = false;
-    Robot.shooterLimelight.setHumanPipelineActive();
+  public void setVisionShootingEnabled(boolean enabled) {
```

Comment:
We are probably going to want to expose a method to get if the shooter is in vision shooting mode. I would also consider moving vision shooting mode to robot state since we have other subsystem specific state values out there (drive type, drive target lock).

Reply:
Agreed. I was going to sit down and plan out the entire robot state soon (hopefully this weekend) and then implement it all once we have all of the main subsystems (shooter, intake, drive) fully made and into the main branch. Any other changes to the RobotState flow will be small enough to do later like a separate state for automatically climbing.